src/main/java/io/github/crucible/grimoire/Grimoire.java

```diff
@@ -157,26 +182,108 @@ public String getSetupClass() {
 
     @Override
     public void injectData(Map<String, Object> data) {
-        //Search for grim patches on grimoire legacy folder
-        File grimoireFolder = new File((File) data.get("mcLocation"), "grimoire");
-        if (grimoireFolder.exists()) {
-            grimPatchList.addAll(scanForPatches(grimoireFolder, true));
-        }
+        if (this.isDevEnvironment()) { // Just handle what's passed in system properties if we are in dev environment
+            logger.info("It would appear we find ourselves in dev environment; scanning system properties for GrimPatch paths...");
+
+            List<URL> paths = this.getGrimpaths();
+            List<GrimPatch> grimPatches = new ArrayList<>();
 
-        //Search for grim patches on mods folder
-        grimPatchList.addAll(scanForPatches(new File((File) data.get("mcLocation"), "mods"), false));
+            for (URL url : paths) {
+                try {
+                    File coreMod = new File(url.toURI().getPath());
+                    Manifest manifest = null;
 
-        //Load for any patch inside the grimoire resource folder.
-        try {
-            grimPatchList.addAll(scanForPatches(extractResources(ResourceList.getResources(Pattern.compile(".*grimoire/.*\\.jar"), classLoader)), false));
-        } catch (IOException e) {
-            logger.error("Error while trying to load embedded patches.");
-            e.printStackTrace();
+                    if (!coreMod.exists()) {
+                        continue;
+                    }
+
+                    if (coreMod.isDirectory()) {
+                        File manifestFile = new File(coreMod, "META-INF/MANIFEST.MF");
+                        if (manifestFile.exists()) {
+                            FileInputStream stream = new FileInputStream(manifestFile);
+                            manifest = new Manifest(stream);
+                            stream.close();
+                        }
+
+                        List<File> mixinEntries = new ArrayList<>();
+
+                        // Try to detect directory grimpatch
+                        for (File file : coreMod.listFiles()) {
+                            if (file.isFile() && this.isMixinConfiguration(file.getName(), false, true)) {
```

Comment:
Same as this https://github.com/juanmuscaria/ForgeGradle/pull/2#discussion_r629462000